lowatt_grdf/main.py

```diff
@@ -69,18 +71,63 @@ def decorator(func: Callback) -> Callback:
                 kwargs["help"] = field.field_info.description
             if field.type_ == bool:
                 kwargs["is_flag"] = True
+            if field.alias.startswith("date_"):
+                if field.alias == "date_consentement_declaree":
+                    # Expect a datetime while it seems more consistent to use a
+                    # simple date.
+                    kwargs["callback"] = _validate_date_as_datetime
+                else:
+                    kwargs["callback"] = _validate_date
+                kwargs["metavar"] = "YYYY-MM-DD"
             click.option(opt, **kwargs)(func)
         return func
 
     return decorator
 
 
-@click.group()
+def _validate_date(ctx: Any, param: Any, value: str) -> str:
+    assert isinstance(value, str), type(value)
+    try:
+        time.strptime(value, "%Y-%m-%d")
+        return value
+    except ValueError as exc:
+        raise click.BadParameter("format must be 'YYYY-MM-DD'") from exc
+
+
+def _validate_date_as_datetime(ctx: Any, param: Any, value: str) -> str:
+    _validate_date(ctx, param, value)
+    return value + " 00:00:00"
+
+
 def main() -> None:
+    try:
+        _main()
+    except requests.HTTPError as exc:
+        LOGGER.error(exc)
+        if exc.response is not None:
+            try:
```

Comment:
Maybe just log exc.response as string is enough ?

Reply:
less readable but probably enough :p